src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private double base;
+    private double height;
+    private double hypotenuse;
+
+    public RightTriangle(Color color, double base, double height) {
+        super(color);
+        this.base = base;
+        this.height = height;
+        this.hypotenuse = Math.sqrt(base * base + height * height);
+    }
+
+    @Override
+    public double getArea() {
+        return 0.5 * base * height;
+    }
+
+    @Override
+    public void print() {
+        System.out.println("Figure: right triangle, area: " + getArea()
+                + " sq. units, base: " + base
+                + " units, height: " + height + " units, hypotenuse: " + hypotenuse
+                + " units, color: " + new ColorSupplier().getRandomColor());

```

Comment:
Use the 'color' field from the superclass 'Figure' instead of generating a new random color in the 'print' method. This ensures that the color is consistent with the one assigned at the creation of the object.